ambry-rest/src/main/java/com.github.ambry.rest/Http2ChannelInitializer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.rest;
+
+import com.github.ambry.commons.SSLFactory;
+import com.github.ambry.config.SSLConfig;
+import com.github.ambry.network.RequestInfo;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.pool.AbstractChannelPoolHandler;
+import io.netty.handler.codec.http2.Http2FrameCodecBuilder;
+import io.netty.handler.codec.http2.Http2MultiplexHandler;
+import io.netty.handler.codec.http2.Http2Settings;
+import io.netty.handler.ssl.SslHandler;
+
+
+/**
+ * A {@link ChannelInitializer} to be used with {@link Http2ChannelPoolMap}.
+ */
+public class Http2ChannelInitializer extends AbstractChannelPoolHandler {
```

Comment:
Could you call this `Http2ClientChannelInitializer` and replace the one used for `Http2BlockingChannel` with this one?

Reply:
Http2ClientChannelInitializer is `Channelnitializer`, which is not a `AbstractChannelPoolHandler`.
We probably need to keep two classes. 
I can rename this(Http2ChannelInitializer) to `Http2ChannelPoolHanlder` and try to avoid dup code.

Reply:
Http2ClientChannelInitializer class is removed. Do inline initialize in Http2BlockingChannel::channel().